src/components/side-navigation/side-navigation.service.ts

```diff
@@ -0,0 +1,10 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+
+@Injectable()
+export class SideNavigationService {
+
+    deselect: Subject<void> = new Subject<void>();

```

Comment:
I think that the Angular router should provide the necessary observables to implement selection/deselection, so I don't think that this service is required. I think what is needed is some kind of provider interface to allow us to plug in to the router, and also use other sources of navigation as required, e.g. a dummy router for the purposes of example code. 

I'll add Ashley as a reviewer, since he has implemented something similar to this recently, and will hopefully be able to give you some explicit guidance.

Reply:
This service likely can be removed now. In a real application the user would be using Angular's routerLink & routerLinkActive directives, which essentially just adds a specified class to an element when it's route is active, eg:

```html
 <ux-side-navigation-item [expanded]="true" header="Menu Levels" icon="hpe-cluster">

    <ux-side-navigation-item header="Second Level">

        <ux-side-navigation-item header="Third Level Item" routerLink="item1" routerLinkActive="active"></ux-side-navigation-item>

    </ux-side-navigation-item>

</ux-side-navigation>
```

We wouldn't want to tie this component directly to the router as they may not always want a click event to perform a navigation, instead they may want to call a function or show a popover for example.

It would probably be best to start by creating a plunker for the section that actually uses the Angular router and use the routerLink and routerLinkActive directives as it would most commonly in a real application. The code to do that should be documented. 

In the example we show in our docs site we can't use the router, but it is easy enough to fake this by simply adding a, `active` class to one of the menu items when they click on it.

Reply:
`routerLinkActive` would also need to control whether the nav item and its ancestors are expanded. So it might be advisable to provide an integration.

Reply:
Yes, it can be injected into a constructor or the menu item and marked as optional, so it should work both with and without the router being used.

```
constructor(@Optional() routerLinkActive: RouterLinkActive) {
}
```